File: src/main/java/net/jcflorezr/util/JsonUtils.java
package net.jcflorezr.util;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.util.Map;

public class JsonUtils {

    private static final ObjectMapper MAPPER = new ObjectMapper();

    public static <T> T convertMapToPojo (Map map, Class<T> pojoClass) {
        return MAPPER.convertValue(map, pojoClass);
    }


}

File: src/main/kotlin/net/jcflorezr/util/JsonUtils.kt
package net.jcflorezr.util

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.registerKotlinModule

object JsonUtils {

    private val MAPPER = ObjectMapper().registerKotlinModule()

    fun <T> convertMapToPojo(map: Map<*, *>, pojoClass: Class<T>) = MAPPER.convertValue(map, pojoClass)!!

    fun convertMapToJsonAsString(map: Map<*, *>) = MAPPER.writeValueAsString(map)

}